.github/workflows/algolia/src/index.ts

```diff
@@ -121,9 +84,22 @@ export const uploadAlgoliaData = async (records) => {
     process.env.AGOLIA_WRITE_KEY
   );
 
-  const index = client.initIndex("hash_learn");
+  const index = client.initIndex("hash_learn_testing1");
 
   return await index
     .saveObjects(records, { autoGenerateObjectIDIfNotExist: true })
     .catch(console.error);
 };
+
+export const deleteDocsIndex = async () => {
+  const client = algoliasearch(
+    process.env.ALGOLIA_PROJECT,
+    process.env.AGOLIA_WRITE_KEY
+  );
+
+  const index = client.initIndex("hash_learn_testing1");
+
+  return await index
+    .deleteBy({ filters: "type:docs OR type:glossary" })
```

Comment:
Filter by type `docs` or `glossary` to delete old index.

Reply:
With the new approach, we delete all objects re-create them again, which is somewhat discouraged by Algolia (see docs for [`deleteBy`](https://www.algolia.com/doc/api-reference/api-methods/delete-by/)). How about calling [`browseIndex`](https://www.algolia.com/doc/api-reference/api-methods/browse/) to get all object ids followed by [`deleteObjects`](https://www.algolia.com/doc/api-reference/api-methods/delete-objects/) for specific ids? If can fetch all objects first, we can diff them with local versions and only save / delete a few objects instead of all.

In theory, this can be postponed till we hit [rate limits](https://support.algolia.com/hc/en-us/articles/4406975251089-Is-there-a-rate-limit-) (not sure how soon we’ll get close to them with a full reset).